### PR TITLE
*: rename Arrow to Chunk

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve? 

In tidb, use `arrow` instead of `chunk` in some places. This can be misleading.

### What is changed and how it works?

In this pr, I change all `arrow` to `chunk`.

### Check List 

Tests 

Code changes

Side effects

Related changes

Release note

### Related PR
https://github.com/pingcap/tipb/pull/151

### CODE REVIEW

File: sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -124,8 +124,8 @@ const (
 	// tidb_enable_streaming enables TiDB to use streaming API for coprocessor requests.
 	TiDBEnableStreaming = "tidb_enable_streaming"
 
-	// tidb_enable_arrow enables TiDB to use Chunk format for coprocessor requests.
-	TiDBEnableArrow = "tidb_enable_arrow"
+	// tidb_enable_chunk enables TiDB to use Chunk format for coprocessor requests.
+	TiDBEnableChunk = "tidb_enable_chunk"
```
Comment (XuHuaiyu) [inline sessionctx/variable/tidb_vars.go]: Will `tidb_enable_chunk_response` be more clearer?

**Comment by XuHuaiyu** [comment]
PTAL @zz-jason

File: config/config.toml.example [hunk 2]
```diff
@@ -283,8 +283,8 @@ max-batch-wait-time = 0
 # Batch wait size, to avoid waiting too long.
 batch-wait-size = 8
 
-# Enable chunk encoded data.
-enable-arrow = true
+# Enable chunk encoded data for coprocessor requests.
+enable-chunk-response = true
```
Comment (zz-jason) [inline config/config.toml.example]: how about s/enable-chunk-response/enable-chunk-rpc/?

File: distsql/distsql.go [hunk 3]
```diff
@@ -152,18 +152,18 @@ func Checksum(ctx context.Context, client kv.Client, kvReq *kv.Request, vars *kv
...
+	if enableTypeChunk(ctx) {
+		dagReq.EncodeType = tipb.EncodeType_TypeChunk
 	} else {
 		dagReq.EncodeType = tipb.EncodeType_TypeDefault
 	}
 }
 
-func enableTypeArrow(ctx sessionctx.Context) bool {
-	if !ctx.GetSessionVars().EnableArrow {
+func enableTypeChunk(ctx sessionctx.Context) bool {
```
Comment (zz-jason) [inline distsql/distsql.go]: how about s/enableTypeChunk/canUseChunkRPC/

File: sessionctx/variable/tidb_vars.go [hunk 4]
```diff
@@ -124,8 +124,8 @@ const (
 	// tidb_enable_streaming enables TiDB to use streaming API for coprocessor requests.
 	TiDBEnableStreaming = "tidb_enable_streaming"
 
-	// tidb_enable_arrow enables TiDB to use Chunk format for coprocessor requests.
-	TiDBEnableArrow = "tidb_enable_arrow"
+	// tidb_enable_chunk_response enables TiDB to use Chunk format for coprocessor requests.
+	TiDBEnableChunkResponse = "tidb_enable_chunk_response"
```
Comment (zz-jason) [inline sessionctx/variable/tidb_vars.go]: ditto

File: util/chunk/chunk.go [hunk 5]
```diff
@@ -25,8 +25,7 @@ import (
 
 var msgErrSelNotNil = "The selection vector of Chunk is not nil. Please file a bug to the TiDB Team"
 
-// Chunk stores multiple rows of data in Apache Arrow format.
-// See https://arrow.apache.org/docs/memory_layout.html
+// Chunk stores multiple rows of data.
```
Comment (zz-jason) [inline util/chunk/chunk.go]: no need to remove the link. Actually, the internal memory layout is the same with apache arrow.

File: util/chunk/chunk_test.go [hunk 6]
```diff
@@ -100,7 +100,7 @@ func (s *testChunkSuite) TestChunk(c *check.C) {
 	row := chk.GetRow(0)
 	c.Assert(row.GetFloat32(0), check.Equals, f32Val)
 	c.Assert(row.GetTime(2).Compare(tVal), check.Equals, 0)
-	// fsp no longer maintain in arrow
+	// fsp no longer maintain in chunk
```
Comment (zz-jason) [inline util/chunk/chunk_test.go]:
```suggestion
	// fsp is no longer maintained in chunk
```

File: util/chunk/column.go [hunk 7]
```diff
@@ -55,8 +55,7 @@ func (c *Column) AppendSet(set types.Set) {
 	c.appendNameValue(set.Name, set.Value)
 }
 
-// Column stores one column of data in Apache Arrow format.
-// See https://arrow.apache.org/docs/memory_layout.html
+// Column stores one column of data.
```
Comment (zz-jason) [inline util/chunk/column.go]: ditto

**Pull request merged**